### ISSUE TITLE
Missing convertions

### ISSUE THREAD

**Issue by mistoll** [ros/geometry2#116]
Once understood the convert method is quite helpfull. But IMO there are some basic convertion methods missing. 

For example:
tf2_geometry_msgs.h missing Vector3, Pose, Point (only stamped versions available)
tf2_eigen.h missing Quaterniond, Matrix3d

Furthermore the documentation on creating convertion methods is incomplete or hard to find.

**Comment by vrabaud** [comment]
right, this documentation needs advertisement, like on the wiki.
You mention a few types, but what would they convert to ? Thx

**Comment by mistoll** [comment]
Nice convertions would be:
Eigen::Quaterniond <=> geometry_msgs::Quaternion
Eigen::Vector3 <=> geometry_msgs::Vector, geometry_msgs::Point
Eigen::Matrix3d <=> geometry_msgs::Matrix
Also great (but might be incompatible)
Eigen::Matrixxd <=> geometry_msgs covariances (e.g. float64[36])

**Comment by vrabaud** [comment]
Ok, if you have any other types that you'd like to have converted, please add in here.

Eigen::Vector3 <=> geometry_msgs::Vector, geometry_msgs::Point

is a tricky one: a vector is not the same things as a point in ROS (a vector is a direction, that is therefore not translated when a transform is applied to it, cf https://github.com/ros/geometry_experimental/blob/jade-devel/tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.h#L77 )

Same with the matrix and the covariance matrix: the info is the same, but not the semantic (same with pose and transform).

Anyway, I need to write the wiki page about conversion and utils.h, and add those conversions above that make sense.

**Comment by mistoll** [comment]
I wasn't aware of the doTransform method. That complicates things.

So what do you recommend for those non-bijective mappings? Should everyone implement them application-specific or not use convert at all?

**Comment by tfoote** [comment]
If it's not bijective it shouldn't use the convert method. You should
define a method which captures the semantic meaning, potentially
application specific, potentially somewhat generic and shareable between
similar applications as any function can be done already.

**Comment by cmansley** [comment]
I agree with @tfoote, but we should define which conversions are the bijective ones and which are not. Because `geometry_msgs::Pose` is being converted into `tf2::Transform` in tf2_geometry_msgs.h, but it could have easily been `geometry_msgs::Transform` instead. 

And these conversion functions are replicated in the buffer_core.cpp.

So, I guess there are two things. 
- When do we use the convert interface defined in convert.h?
- What is the policy for defining "custom" methods?

**Comment by tfoote** [comment]
#202 adds the primatives for geometry_msgs

**Comment by sloretz** [comment]
ROS 1 is end-of-life (EOL) as of today, May 31st 2025. I am archiving this repository because:

* it only supports ROS 1
* it isn't needed anymore in ROS 2

[If you still rely on ROS 1, read this page to learn about your options.](https://bit.ly/NoeticEOL)
